src/Groups/GAPGroups.jl

```diff
@@ -372,9 +372,26 @@ end
 
 Base.:*(x::GAPGroupElem, y::GAPGroupElem) = _prod(x, y)
 
-==(x::GAPGroup, y::GAPGroup) = GapObj(x) == GapObj(y)
+function ==(x::GAPGroup, y::GAPGroup)
+  _check_compatible(x, y; error = false) || throw(ArgumentError("x and y are not compatible"))
+  return GapObj(x) == GapObj(y)
+end
+
+# For two `BasicGAPGroupElem`s,
+# we allow the question for equality if their parents fit together
+# in the sense of `_check_compatible`,
+# and compare the `GapObj`s if this is the case.
+function ==(x::BasicGAPGroupElem, y::BasicGAPGroupElem)
+  _check_compatible(parent(x), parent(y); error = false) || throw(ArgumentError("parents of x and y are not compatible"))
+  return GapObj(x) == GapObj(y)
+end
 
-==(x::BasicGAPGroupElem, y::BasicGAPGroupElem ) = GapObj(x) == GapObj(y)
+# For two `GAPGroupElem`s,
+# if no specialized method is applicable then no `==` comparison is allowed.
+function ==(x::GAPGroupElem, y::GAPGroupElem)
+  _check_compatible(parent(x), parent(y); error = false) || throw(ArgumentError("parents of x and y are not compatible"))
+  throw(ArgumentError("== is not implemented for the given types"))
+end
```

Comment:
once https://github.com/Nemocas/AbstractAlgebra.jl/pull/1853 is available, this method can be removed. maybe add this as a comment

Reply:
My point of view is:
If the compatibility check returns `false` then we are sure that the question for `==` is not allowed,
but if the compatibility check returns `true` then likely there is a new subtype of `GAPGroupElem` for which a `==` method should be provided.
(Currently we have just the subtypes `BasicGAPGroupElem` and `MatrixGroupElem`.)